vars/cloudFoundryCreateService.groovy

```diff
@@ -1,180 +1,11 @@
-         * observed when using `cf push --var` in combination with `cf push --vars-file`.
-         */
-        'manifestVariables',
-        /**
-         * Cloud Foundry target organization.
-         * @parentConfigKey cloudFoundry
-         */
-        'org',
-        /**
-         * Cloud Foundry target space.
-         * @parentConfigKey cloudFoundry
-         */
-        'space',
-    /** @see dockerExecute */
-    'dockerImage',
-    /** @see dockerExecute */
-    'dockerWorkspace',
-    /** @see dockerExecute */
-    'stashContent'
-]
+@Field String STEP_NAME = getClass().getName()
+@Field String METADATA_FILE = 'metadata/cloudFoundryCreateService.yaml'
 
-@Field Map CONFIG_KEY_COMPATIBILITY = [cloudFoundry: [apiEndpoint: 'cfApiEndpoint', appName:'cfAppName', credentialsId: 'cfCredentialsId', serviceManifest: 'cfServiceManifest', manifestVariablesFiles: 'cfManifestVariablesFiles', manifestVariables: 'cfManifestVariables',  org: 'cfOrg', space: 'cfSpace']]
-@Field Set GENERAL_CONFIG_KEYS = STEP_CONFIG_KEYS
-@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
-
-/**
- * Step that uses the CF Create-Service-Push plugin to create services in a Cloud Foundry space. The information about the services is provided in a yaml file as infrastructure as code.
- * It is possible to use variable substitution inside of the yaml file like in a CF-push manifest yaml.
- *
- * For more details how to specify the services in the yaml see the [github page of the plugin](https://github.com/dawu415/CF-CLI-Create-Service-Push-Plugin).
- *
- * The `--no-push` options is always used with the plugin. To deploy the application make use of the cloudFoundryDeploy step!
- */
-@GenerateDocumentation
 void call(Map parameters = [:]) {
-    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
-        def script = checkScript(this, parameters) ?: this
-        def utils = parameters.juStabUtils ?: new Utils()
-        def jenkinsUtils = parameters.jenkinsUtilsStub ?: new JenkinsUtils()
-        // load default & individual configuration
-        Map config = ConfigurationHelper.newInstance(this)
-            .loadStepDefaults()
-            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
-            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
-            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
-            .mixin(parameters, PARAMETER_KEYS, CONFIG_KEY_COMPATIBILITY)
-            .withMandatoryProperty('cloudFoundry/org')
-            .withMandatoryProperty('cloudFoundry/space')
-            .withMandatoryProperty('cloudFoundry/credentialsId')
-            .withMandatoryProperty('cloudFoundry/serviceManifest')
-            .use()
-
-
-        utils.pushToSWA([step: STEP_NAME],config)
-
-        utils.unstashAll(config.stashContent)
```

Comment:
This seems to be missing in the go implementation. Or is this handled differently?

Reply:
seems to be handled implicitly in piperExecuteBin